core/src/test/java/apoc/trigger/TriggerClusterRoutingTest.java

```diff
@@ -0,0 +1,102 @@
+package apoc.trigger;
+
+import apoc.util.Neo4jContainerExtension;
+import apoc.util.TestContainerUtil;
+import apoc.util.TestcontainersCausalCluster;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.neo4j.driver.Session;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static apoc.trigger.Trigger.SYS_NON_LEADER_ERROR;
+import static apoc.trigger.TriggerNewProcedures.TRIGGER_NOT_ROUTED_ERROR;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class TriggerClusterRoutingTest {
+
+    private static TestcontainersCausalCluster cluster;
+
+    @BeforeClass
+    public static void setupCluster() {
+        cluster = TestContainerUtil
+                .createEnterpriseCluster(3, 1, Collections.emptyMap(), Map.of(
+                        "NEO4J_dbms_routing_enabled", "true",
+                        "apoc.trigger.enabled", "true"
+                ));
+    }
+
+    @AfterClass
+    public static void bringDownCluster() {
+        if (cluster != null) {
+            cluster.close();
+        }
+    }
+
+    // TODO: making sure that a session against "system" can install triggers
+
+    // TODO: making sure that a session against "system" can drop triggers
+    
+    // TODO: fabric tests
+
+    @Test
+    public void testTriggerInstallAllowedOnlyInSysLeaderMember() {
+        final String query = "CALL apoc.trigger.install('neo4j', $name, 'RETURN 1',{})";
+        triggerInSysLeaderMemberCommon(query, TRIGGER_NOT_ROUTED_ERROR);
+    }
+
+    @Test
+    public void testTriggerAddAllowedOnlyInSysLeaderMember() {
+        final String query = "CALL apoc.trigger.add($name, 'RETURN 1',{})";
+        triggerInSysLeaderMemberCommon(query, SYS_NON_LEADER_ERROR);
+    }
+
+    @Test
+    public void testTriggerRemoveAllowedOnlyInSysLeaderMember() {
+        final String query = "CALL apoc.trigger.remove('neo4j', $name)";
+        triggerInSysLeaderMemberCommon(query, TRIGGER_NOT_ROUTED_ERROR);
+    }
+
+    @Test
+    public void testTriggerDropAllowedOnlyInSysLeaderMember() {
+        final String query = "CALL apoc.trigger.drop('neo4j', $name)";
+        triggerInSysLeaderMemberCommon(query, SYS_NON_LEADER_ERROR);
+    }
+
+    private static void triggerInSysLeaderMemberCommon(String query, String triggerNotRoutedError) {
+        final List<Neo4jContainerExtension> members = cluster.getClusterMembers();
+        assertEquals(4, members.size());
+        for (Neo4jContainerExtension container: members) {
+            // we skip READ_REPLICA members
+            final String readReplica = TestcontainersCausalCluster.ClusterInstanceType.READ_REPLICA.toString();
+            try (final Session session = container.getSession()) {
```

Comment:
I'm not sure you can do this. Wouldn't this try and close the session, which is a variable inside the container?

I would leave this without the `try`